account_asset_management/models/account_asset.py

```diff
@@ -646,35 +654,75 @@ def _compute_depreciation_line(
                     seq -= 1
             line_i_start = 0
 
-    def compute_depreciation_board(self):
+    def _unlink_asset_lines_whitout_move(self):
+        self.ensure_one()
+        line_obj = self.env["account.asset.line"]
+        domain = [
+            ("asset_id", "=", self.id),
+            ("type", "=", "depreciate"),
+            ("move_id", "=", False),
+            ("init_entry", "=", False),
+        ]
+        line_obj.search(domain).unlink()
 
+    def _get_asset_posted_lines(self):
+        self.ensure_one()
         line_obj = self.env["account.asset.line"]
+        domain = [
+            ("asset_id", "=", self.id),
+            ("type", "=", "depreciate"),
+            "|",
+            ("move_check", "=", True),
+            ("init_entry", "=", True),
+        ]
+        posted_lines = line_obj.search(domain, order="line_date desc")
+
+        last_line = line_obj
+        if posted_lines:
+            last_line = posted_lines[0]
+        return posted_lines, last_line
+
+    def _add_amount_diff_to_next_deprecation_lines(
```

Comment:
This method is ugly due to all the parameters involved and don't understand why you need to extract it, as it's only call once and don't require to be overridden.

Reply:
I passed this on to two methods for reducing complexity.

account_asset_management/models/account_asset.py:657:5: C901 'AccountAsset.compute_depreciation_board' is too complex (19)

Reply:
![Selección_760](https://github.com/OCA/account-financial-tools/assets/6359121/62affd90-bc2c-4f20-86cb-918f01c4aa15)